src/common/Preforker.h

```diff
@@ -108,7 +108,8 @@ class Preforker {
   int signal_exit(int r) {
     if (forked) {
       /* If we get an error here, it's too late to do anything reasonable about it. */
-      (void)safe_write(fd[1], &r, sizeof(r));
+      ssize_t len = safe_write(fd[1], &r, sizeof(r));
```

Comment:
could just put
```c++
[[maybe_unused]] auto n = safe_write(fd[1], &r, sizeof(r));
```

Reply:
while testing this change, i spotted another warning. so i am trying to fix them in #21484. could you help take a look at it?